sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalog.scala

```diff
@@ -261,6 +261,20 @@ trait ExternalCatalog {
       table: String,
       partialSpec: Option[TablePartitionSpec] = None): Seq[CatalogTablePartition]
 
+  /**
+   * List the metadata of all partitions that belong to the specified table
+   * and the giving part names,assuming it exists.
+   *
+   * @param db database name
+   * @param table table name
+   * @param partitionNames partNames e.g.  ('a=1/b=1', 'a=1/b=2')
+   * @return
+   */
+  def listPartitionsByNames(
+      db: String,
+      table: String,
+      partNames: Seq[String]): Seq[CatalogTablePartition]
```

Comment:
Why not still use `TablePartitionSpec` here ?

Reply:
Because if using TablePartitionSpec, functionally, it is similar to the existing listPartitions, not friendly to retrieve multiple partitions that are already determined.

Reply:
How about `partialSpecs: Seq[TablePartitionSpec]` ?

Reply:
Good point, but the behavior will be different from before. For example, the original `listPartitions` can support partial partition input (e.g. `Map(a->1)`) and match it with partitions like `(a=1/b=1)` and `(a=1/b=2)`. The input of list performs an exact match, so only an input of `List(map(a->1, b->2))` will match the partition `(a=1/b2=2)`. Is this acceptable?

Reply:
I got it.